### ISSUE TITLE
Post training quantization conversion log show: Optimization loop failed: Cancelled: Operation was cancelled

### ISSUE THREAD

**Issue by Ethylbenzol** [tensorflow/tensorflow#48689]
### 1. System information

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- TensorFlow installation (pip package or built from source): pip package install in anaconda virtual env
- TensorFlow library (version, if pip package or github SHA, if built from source):  tf-nightly 2.6.0.dev20210407

### 2. Code

```
import numpy as np
import tensorflow as tf

def representative_dataset():
    path = r'D:\Development\fruits_data\images\test'
    image_dataset = tf.data.Dataset.list_files(path + '/*.png')
    for i in range(15):
        image = next(iter(image_dataset))
        image = tf.io.read_file(image)
        image = tf.io.decode_jpeg(image, channels=3)
        image = tf.image.resize(image, [640, 640])
        image = tf.cast(image / 255., tf.float32)
        image = tf.expand_dims(image, 0)
        yield [image]
        
#import trained model from mobilenet 640 v2 fpn
converter = tf.lite.TFLiteConverter.from_saved_model('./exported_models/tflite/saved_model') #using tensorflow

#quantize the model to int8 for export
converter.experimental_new_converter = True #required for using experimental target_spec flags.
converter.optimizations = [tf.lite.Optimize.DEFAULT] #Currently only DEFAULT is supported by TF2.4.X or tf-nightly.
converter.representative_dataset = representative_dataset
converter.target_spec.supported_ops = [tf.lite.OpsSet.TFLITE_BUILTINS_INT8]

converter.target_spec.supported_types = [tf.int8]
converter.inference_input_type = tf.int8 
#converter.inference_output_type = tf.int8 
converter.allow_custom_ops = True 
tflite_int8_model = converter.convert()

# Save the model.
with open('test_model.tflite', 'wb') as f:
  f.write(tflite_int8_model)
```

### 3. Failure after conversion
If the conversion is successful, but the generated model is wrong, then state what is wrong:

Conversion to TFLite is successful, but on the log states: `Optimization loop failed: Cancelled: Operation was cancelled`
and during inferences the model was not able to be used. The conversion model used is the pre-trained `mobilenet 640 v2 fpn` model found in tf zoo for object detection. 

### 4. (optional) RNN conversion support
~If converting TF RNN to TFLite fused RNN ops, please prefix [RNN] in the title.~

### 5. (optional) Any other info / logs

Terminal log: 

```
Skipping registering GPU devices...
2021-04-22 12:08:07.890584: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1258] Device interconnect StreamExecutor with strength 1 edge matrix:
2021-04-22 12:08:07.890670: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1264]      0
2021-04-22 12:08:07.891405: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1277] 0:   N
2021-04-22 12:08:09.301186: W tensorflow/core/kernels/data/model_dataset_op.cc:205] Optimization loop failed: Cancelled: Operation was cancelled
2021-04-22 12:08:10.346812: W tensorflow/core/kernels/data/model_dataset_op.cc:205] Optimization loop failed: Cancelled: Operation was cancelled
2021-04-22 12:08:11.392280: W tensorflow/core/kernels/data/model_dataset_op.cc:205] Optimization loop failed: Cancelled: Operation was cancelled
2021-04-22 12:08:12.466013: W tensorflow/core/kernels/data/model_dataset_op.cc:205] Optimization loop failed: Cancelled: Operation was cancelled
2021-04-22 12:08:13.526607: W tensorflow/core/kernels/data/model_dataset_op.cc:205] Optimization loop failed: Cancelled: Operation was cancelled
2021-04-22 12:08:14.586672: W tensorflow/core/kernels/data/model_dataset_op.cc:205] Optimization loop failed: Cancelled: Operation was cancelled
2021-04-22 12:08:15.633286: W tensorflow/core/kernels/data/model_dataset_op.cc:205] Optimization loop failed: Cancelled: Operation was cancelled
2021-04-22 12:08:16.680019: W tensorflow/core/kernels/data/model_dataset_op.cc:205] Optimization loop failed: Cancelled: Operation was cancelled
2021-04-22 12:08:17.754911: W tensorflow/core/kernels/data/model_dataset_op.cc:205] Optimization loop failed: Cancelled: Operation was cancelled
2021-04-22 12:08:18.828891: W tensorflow/core/kernels/data/model_dataset_op.cc:205] Optimization loop failed: Cancelled: Operation was cancelled
2021-04-22 12:08:19.905723: W tensorflow/core/kernels/data/model_dataset_op.cc:205] Optimization loop failed: Cancelled: Operation was cancelled
2021-04-22 12:08:20.984154: W tensorflow/core/kernels/data/model_dataset_op.cc:205] Optimization loop failed: Cancelled: Operation was cancelled
2021-04-22 12:08:22.027698: W tensorflow/core/kernels/data/model_dataset_op.cc:205] Optimization loop failed: Cancelled: Operation was cancelled
2021-04-22 12:08:23.097239: W tensorflow/core/kernels/data/model_dataset_op.cc:205] Optimization loop failed: Cancelled: Operation was cancelled
```

**Comment by abattery** [comment]
@liufengdb could you triage this?

**Comment by liufengdb** [comment]
Ethylbenzol@, can you use some random inputs in your representative_dataset method to verify that error is actually due to the use of tf.dataset (not really quantization).

**Comment by Ethylbenzol** [comment]
>  can you use some random inputs in your representative_dataset method to verify that error is actually due to the use of tf.dataset (not really quantization).

'''
def representative_dataset():
    for _ in range(100):
      data = np.random.rand(1, 640, 640, 3)
      yield [data.astype(np.float32)]
''
so i ran this dummy dataset and seem like there was no operation cancelled error.

**Comment by Ethylbenzol** [comment]
update to this issue:
Looking at the logs during tflite inference for this model: 
```
{'name': 'StatefulPartitionedCall:3', 'index': 382, 'shape': array([ 1, 10,  4]), 'shape_signature': array([ 1, 10,  4]), 'dtype': <class 'numpy.uint8'>, 'quantization': (0.006796886213123798, 60), 'quantization_parameters': {'scales': array([0.00679689], dtype=float32), 'zero_points': array([60]), 'quantized_dimension': 0}, 'sparsity_parameters': {}}
{'name': 'StatefulPartitionedCall:2', 'index': 384, 'shape': array([ 1, 10]), 'shape_signature': array([ 1, 10]), 'dtype': <class 'numpy.uint8'>, 'quantization': (0.01568627543747425, 0), 'quantization_parameters': {'scales': array([0.01568628], dtype=float32), 'zero_points': array([0]), 'quantized_dimension': 0}, 'sparsity_parameters': {}}
{'name': 'StatefulPartitionedCall:1', 'index': 386, 'shape': array([ 1, 10]), 'shape_signature': array([ 1, 10]), 'dtype': <class 'numpy.uint8'>, 'quantization': (0.0035720604937523603, 0), 'quantization_parameters': {'scales': array([0.00357206], dtype=float32), 'zero_points': array([0]), 'quantized_dimension': 0}, 'sparsity_parameters': {}}
{'name': 'StatefulPartitionedCall:0', 'index': 388, 'shape': array([1]), 'shape_signature': array([1]), 'dtype': <class 'numpy.uint8'>, 'quantization': (0.03921568766236305, 0), 'quantization_parameters': {'scales': array([0.03921569], dtype=float32), 'zero_points': array([0]), 'quantized_dimension': 0}, 'sparsity_parameters': {}}
this is the detection_boxes: [[ 71  86 216 204]
 [ 72  34 221 227]
 [166 200 191 208]
 [ 46   0 197 212]
 [ 95   2 245 209]
 [180  58 208  66]
 [156 191 201 211]
 [158 199 191 206]
 [ 91 203 113 207]
 [172  60 194  63]] and this is the detection_classes: and this is the detection_scores: [ 64  64 255  64  64 255 255   0 255 255] and this is the count: [236  28   5   5   5   5   3   3   2   2] 
```

Seem like there the detection_classes are missing and the score make no sense. Are there anything else i could provide to troubleshoot this?

**Comment by yeshbourne** [comment]
import numpy as np
import tensorflow as tf

image_data = tf.data.Dataset.list_files("./quant-images/*.jpg")
HEIGHT, WIDTH = 640, 640

def representative_dataset_gen():
   for image_path in image_data:
       img = tf.io.read_file(image_path)
       img = tf.io.decode_image(img, channels=3)
       img = tf.image.convert_image_dtype(img, tf.float32)
       resized_img = tf.image.resize(img, (HEIGHT, WIDTH))
       resized_img = resized_img[tf.newaxis, :]
       yield [resized_img]

model_path = './exported-models/ptag-detector-model/ssd_mobilenet_v2_fpnlite_640x640/saved_model'        

#import trained model from mobilenet 640 v2 fpn
converter = tf.lite.TFLiteConverter.from_saved_model(model_path) #using tensorflow

converter.representative_dataset = representative_dataset_gen
converter.target_spec.supported_ops = [tf.lite.OpsSet.TFLITE_BUILTINS_INT8]
converter.inference_input_type = tf.uint8
converter.inference_output_type = tf.uint8
converter.quantized_input_stats = {"normalized_input_image_tensor": (128, 128)}
converter.optimizations = [tf.lite.Optimize.DEFAULT]

tflite_model_quant = converter.convert()

interpreter = tf.lite.Interpreter(model_content=tflite_model_quant)
input_type = interpreter.get_input_details()[0]['dtype']
print('input: ', input_type)
output_type = interpreter.get_output_details()[0]['dtype']
print('output: ', output_type)

# Save the model.
with open('detect_quant.tflite', 'wb') as f:
  f.write(tflite_model_quant)


This code will get you full quantized tflite model
